jerry-core/vm/vm-stack.c

```diff
@@ -189,16 +231,15 @@ vm_stack_find_finally (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
 
   while (frame_ctx_p->context_depth > 0)
   {
-    vm_stack_context_type_t context_type;
+    vm_stack_context_type_t context_type = VM_GET_CONTEXT_TYPE (stack_top_p[-1]);
     uint32_t context_end = VM_GET_CONTEXT_END (stack_top_p[-1]);
 
-    if (search_limit < context_end)
+    if (!VM_CONTEXT_HAS_VARIABLE_LENGTH (context_type) && search_limit < context_end)
```

Comment:
Since `break` and `continue` are statements, it is impossible to encounter a variable length context when they are running. This could be checked by an assert and expolited in some way (e.g. when the search_limit is UINT32_MAX, this check is never true).